pkg/controller/lifecycle/actuator.go

```diff
@@ -695,6 +708,31 @@ func getShootResources(webhookCaBundle []byte, extensionNamespace, shootAccessSe
 	return shootResources, nil
 }
 
+func getClientKeys(ctx context.Context, client client.Client, resources []v1beta1.NamedResourceReference, resourceName, namespace string) ([]byte, error) {
+    ref := v1beta1helper.GetResourceByName(resources, resourceName)
+    if ref == nil || ref.ResourceRef.Kind != "Secret" {
+        return nil, fmt.Errorf("failed to find referenced resource with name %s and kind Secret", resourceName)
+    }
+
+    refSecret := &corev1.Secret{
+        ObjectMeta: metav1.ObjectMeta{
+            Name:      ref.ResourceRef.Name,
+            Namespace: namespace,
+        },
+    }
+
+    if err := controller.GetObjectByReference(ctx, client, &ref.ResourceRef, namespace, refSecret); err != nil {
+        return nil, fmt.Errorf("failed to read referenced secret %s%s for reference %s: %s", v1beta1constants.ReferencedResourcesPrefix, ref.ResourceRef.Name, resourceName, err)
```

Comment:
```suggestion
        return nil, fmt.Errorf("failed to read referenced secret %s%s for reference %s: %w", v1beta1constants.ReferencedResourcesPrefix, ref.ResourceRef.Name, resourceName, err)
```

It is better to use `%w` for errors, this could help up in the stack if the caller want to unwrap the error for some reason.